app/models/status_message.rb

```diff
@@ -128,7 +128,11 @@ def update_and_dispatch_attached_photos(sender)
   end
 
   def comment_email_subject
-    message.title length: 70
+    if self.public?
+      message.title length: 70
+    else
+      I18n.t("status_messages.limited_subject")
```

Comment:
I think `notifier.comment_on_post` would make a better location :)

Also how about moving the "Re: " added at app/mailers/notifications/comment_on_post.rb to a translation and leaving it off here, having the subject something like "There's a new comment on a post you commented" instead of "Re: Limited post"?